packages/utils/src/envelope.ts

```diff
@@ -22,11 +22,18 @@ export function addItemToEnvelope<E extends Envelope>(envelope: E, newItem: E[1]
 }
 
 /**
- * Get the type of the envelope. Grabs the type from the first envelope item.
+ * Convenience function to loop through the items and item types of an envelope.
+ * (This function was mostly created because working with envelope types is painful at the moment)
```

Comment:
on purpose because they are tuple types :P - but very fair

Introducing something like `forEachEnvelopeItem` is actually good behaviour because we intentionally want to treat the envelope data structure as a void pointer (only allow functions to access and mutate it).

Reply:
Interesting. Did not know that - thanks for bringing it up!